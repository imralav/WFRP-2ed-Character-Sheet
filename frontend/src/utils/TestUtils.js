import React, {Fragment} from 'react';

import en from 'react-intl/locale-data/en';
import {addLocaleData, IntlProvider} from 'react-intl';
import renderer from 'react-test-renderer';

import {BrowserRouter as Router, Route} from 'react-router-dom';

addLocaleData(en);

class ComponentBuilder {
    constructor(component) {
        this.component = component;
    }

    withIntl() {
        this.component = (
            <IntlProvider locale="en" messages={{}}>
                <Fragment>
                {this.component}
                </Fragment>
            </IntlProvider>
        );
        return this;
    }

    withRouter() {
        this.component = (
            <Router>
                <Fragment>
                    {this.component}
                </Fragment>
            </Router>
        );
        return this;
    }

    build() {
        return renderer.create(this.component);
    }
}

export default class TestUtils {
    static buildComponent(component) {
        return new ComponentBuilder(component);
    }
}