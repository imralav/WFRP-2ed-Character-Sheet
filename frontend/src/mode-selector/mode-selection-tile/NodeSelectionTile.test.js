import React from 'react';
import NodeSelectionTile from '../mode-selection-tile';
import TestUtils from '../../utils/TestUtils';
import CreateCharacterImg from '../create-character.png';

describe('NodeSelectionTile', () => {
    const SOME_LINK = "/characterSheet";
    const SOME_TEXT = "someText";
    let component;
    let componentRoot;
    let tree;
    let changeLocaleFunction;
    beforeEach(() => {
        changeLocaleFunction = jest.fn();
        component = TestUtils.buildComponent(
            <NodeSelectionTile img={CreateCharacterImg} link={SOME_LINK} text={SOME_TEXT}/>
        ).withIntl().withRouter().build();
        componentRoot = component.root;
        tree = component.toJSON();
    });
    it('should match previous snapshot', () => {
        expect(tree).toMatchSnapshot();
    });
    it('should have correct text', () => {
        expect(componentRoot.findByType('span').props.children).toEqual(SOME_TEXT);
    });
    it('should have correct link', () => {
        expect(componentRoot.findByType('a').props.href).toEqual(SOME_LINK);
    });
});
