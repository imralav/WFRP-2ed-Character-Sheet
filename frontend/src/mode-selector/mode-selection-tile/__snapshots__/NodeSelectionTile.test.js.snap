// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeSelectionTile should match previous snapshot 1`] = `
Array [
  <a
    className="modeSelectorTile text-center border border-dark rounded d-inline-flex flex-column"
    href="/characterSheet"
    onClick={[Function]}
>
    <div>
        <img
            alt=""
            src="create-character.png"
        />
    </div>
    <div>
        <span>
            someText
        </span>
    </div>
</a>,
  ";",
]
`;
